signature/types.go

```diff
@@ -0,0 +1,133 @@
+package signature
+
+import (
+	"crypto/x509"
+	"errors"
+	"time"
+)
+
+// MediaTypePayloadV1 is the supported content type for signature's payload.
+const MediaTypePayloadV1 = "application/vnd.cncf.notary.payload.v1+json"
+
+// SigningScheme formalizes the feature set (guarantees) provided by
+// the signature.
+// Reference: https://github.com/notaryproject/notaryproject/blob/main/signing-scheme.md
+type SigningScheme string
+
+// SigningSchemes supported by notation.
+const (
+	// notary.x509 signing scheme.
+	SigningSchemeX509 SigningScheme = "notary.x509"
+
+	// notary.x509.signingAuthority schema.
+	SigningSchemeX509SigningAuthority SigningScheme = "notary.x509.signingAuthority"
+)
+
+// SignedAttributes represents signed metadata in the signature envelope.
+// Reference: https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#signed-attributes
+type SignedAttributes struct {
```

Comment:
`authentic signing time` should be in another PR if we need it explicitly since it is not there in the `main` branch: https://github.com/notaryproject/notation-core-go/blob/main/signer/signer.go#L45